lib/matplotlib/colors.py

```diff
@@ -232,9 +232,34 @@ def is_color_like(c):
 
 
 def _has_alpha_channel(c):
-    """Return whether *c* is a color with an alpha channel."""
-    # 4-element sequences are interpreted as r, g, b, a
-    return not isinstance(c, str) and len(c) == 4
+    """Return whether *c* is a color with an alpha channel"""
```

Comment:
We should mention that the result is undefined if *c* is not a valid color specifier.